examples/cuda/hapi-cuda-callback.py

```diff
@@ -0,0 +1,36 @@
+from charm4py import charm
+import time
+import torch
+
+# using numba requires the following stream handle
+# import numba.cuda as cuda
+# s = cuda.stream()
+# stream_handle = s.handle.value
+
+def main(args):        
+    cuda = torch.device('cuda')
+    s = torch.cuda.Stream()  # Create a new stream.
+    A = torch.empty((100, 100), device=cuda).normal_(0.0, 1.0)
```

Comment:
Do you think it would be better if the default here was `numba`? Torch seems a much heavier dependency to test this functionality.